### PR TITLE
[new release] websocket (3 packages) (2.17)

### PR DESCRIPTION
**Pull request by vbmithr**
Websocket library

- Project page: <a href="https://github.com/vbmithr/ocaml-websocket">https://github.com/vbmithr/ocaml-websocket</a>
- Documentation: <a href="https://vbmithr.github.io/ocaml-websocket/doc">https://vbmithr.github.io/ocaml-websocket/doc</a>

##### CHANGES:

- adapt to mirage-crypto (@copy)
- use mirage-crypto-rng as csprng (@ushitora-anqou)

### CODE REVIEW

**Comment by shonfeder** [comment]
Hello! Thanks for the release :)

Looks like a number of failures showing up in CI, beginning with

```
#=== ERROR while compiling websocket.2.17 =====================================#
# context              2.3.0~alpha~dev | linux/x86_64 | ocaml-base-compiler.4.08.1 | file:///home/opam/opam-repository
# path                 ~/.opam/4.08/.opam-switch/build/websocket.2.17
# command              ~/.opam/opam-init/hooks/sandbox.sh build dune build -p websocket -j 31 --promote-install-files=false @install
# exit-code            1
# env-file             ~/.opam/log/websocket-7-8e6f49.env
# output-file          ~/.opam/log/websocket-7-8e6f49.out
### output ###
# (cd _build/default && /home/opam/.opam/4.08/bin/ocamlc.opt -w -40 -g -bin-annot -I core/.websocket.objs/byte -I /home/opam/.opam/4.08/lib/angstrom -I /home/opam/.opam/4.08/lib/astring -I /home/opam/.opam/4.08/lib/base/caml -I /home/opam/.opam/4.08/lib/base64 -I /home/opam/.opam/4.08/lib/bigstringaf -I /home/opam/.opam/4.08/lib/bytes -I /home/opam/.opam/4.08/lib/cohttp -I /home/opam/.opam/4.08/lib/conduit -I /home/opam/.opam/4.08/lib/cstruct -I /home/opam/.opam/4.08/lib/domain-name -I /home/opam/.opam/4.08/lib/eqaf -I /home/opam/.opam/4.08/lib/eqaf/bigstring -I /home/opam/.opam/4.08/lib/eqaf/cstruct -I /home/opam/.opam/4.08/lib/ipaddr -I /home/opam/.opam/4.08/lib/ipaddr-sexp -I /home/opam/.opam/4.08/lib/macaddr -I /home/opam/.opam/4.08/lib/mirage-crypto -I /home/opam/.opam/4.08/lib/mirage-crypto-rng -I /home/opam/.opam/4.08/lib/ocplib-endian -I /home/opam/.opam/4.08/lib/parsexp -I /home/opam/.opam/4.08/lib/ppx_sexp_conv/runtime-lib -I /home/opam/.opam/4.08/lib/re -I /home/opam/.opam/4.08/lib/seq -I /home/opam/.opam/4.08/lib/sexplib -I /home/opam/.opam/4.08/lib/sexplib0 -I /home/opam/.opam/4.08/lib/stringext -I /home/opam/.opam/4.08/lib/uri -I /home/opam/.opam/4.08/lib/uri-sexp -intf-suffix .ml -no-alias-deps -open Websocket__ -o core/.websocket.objs/byte/websocket.cmo -c -impl core/websocket.ml)
# File "core/websocket.ml", line 1:
# Error: The implementation core/websocket.ml
#        does not match the interface core/.websocket.objs/byte/websocket.cmi:
#        In module Rng:
#        Modules do not match:
#          sig val init : unit -> int -> Cstruct.t end
#        is not included in
#          sig val init : unit -> int -> string end
#        In module Rng:
#        Values do not match:
#          val init : unit -> int -> Cstruct.t
#        is not included in
#          val init : unit -> int -> string
#        File "core/websocket.mli", line 25, characters 2-34:
#          Expected declaration
#        File "core/websocket.ml", line 24, characters 6-10: Actual declaration
# (cd _build/default && /home/opam/.opam/4.08/bin/ocamlopt.opt -w -40 -g -I core/.websocket.objs/byte -I core/.websocket.objs/native -I /home/opam/.opam/4.08/lib/angstrom -I /home/opam/.opam/4.08/lib/astring -I /home/opam/.opam/4.08/lib/base/caml -I /home/opam/.opam/4.08/lib/base64 -I /home/opam/.opam/4.08/lib/bigstringaf -I /home/opam/.opam/4.08/lib/bytes -I /home/opam/.opam/4.08/lib/cohttp -I /home/opam/.opam/4.08/lib/conduit -I /home/opam/.opam/4.08/lib/cstruct -I /home/opam/.opam/4.08/lib/domain-name -I /home/opam/.opam/4.08/lib/eqaf -I /home/opam/.opam/4.08/lib/eqaf/bigstring -I /home/opam/.opam/4.08/lib/eqaf/cstruct -I /home/opam/.opam/4.08/lib/ipaddr -I /home/opam/.opam/4.08/lib/ipaddr-sexp -I /home/opam/.opam/4.08/lib/macaddr -I /home/opam/.opam/4.08/lib/mirage-crypto -I /home/opam/.opam/4.08/lib/mirage-crypto-rng -I /home/opam/.opam/4.08/lib/ocplib-endian -I /home/opam/.opam/4.08/lib/parsexp -I /home/opam/.opam/4.08/lib/ppx_sexp_conv/runtime-lib -I /home/opam/.opam/4.08/lib/re -I /home/opam/.opam/4.08/lib/seq -I /home/opam/.opam/4.08/lib/sexplib -I /home/opam/.opam/4.08/lib/sexplib0 -I /home/opam/.opam/4.08/lib/stringext -I /home/opam/.opam/4.08/lib/uri -I /home/opam/.opam/4.08/lib/uri-sexp -intf-suffix .ml -no-alias-deps -open Websocket__ -o core/.websocket.objs/native/websocket.cmx -c -impl core/websocket.ml)
# File "core/websocket.ml", line 1:
# Error: The implementation core/websocket.ml
#        does not match the interface core/.websocket.objs/byte/websocket.cmi:
#        In module Rng:
#        Modules do not match:
#          sig val init : unit -> int -> Cstruct.t end
#        is not included in
#          sig val init : unit -> int -> string end
#        In module Rng:
#        Values do not match:
#          val init : unit -> int -> Cstruct.t
#        is not included in
#          val init : unit -> int -> string
#        File "core/websocket.mli", line 25, characters 2-34:
#          Expected declaration
#        File "core/websocket.ml", line 24, characters 6-10: Actual declaration



<><> Error report <><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><>
+- The following actions failed
| - build websocket 2.17
+- 
```

https://opam.ci.ocaml.org/github/ocaml/opam-repository/commit/3f38a28f9229e51d12d85b68724b7bcabeab7398/variant/compilers,4.08,websocket-lwt-unix.2.17

File: packages/websocket/websocket.2.17/opam [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+bug-reports: "https://github.com/vbmithr/ocaml-websocket/issues"
+depends: [
+  "dune" {>= "2.9"}
+  "ocaml" {>= "4.06.0"}
+  "base64" {>= "3.3.0"}
+  "conduit" {>= "5.1.0"}
+  "cohttp" {>= "5.0.0"}
+  "ocplib-endian" {>= "1.0"}
+  "astring" {>= "0.8.3"}
+  "mirage-crypto-rng"
```
Comment (copy) [inline packages/websocket/websocket.2.17/opam:32]:
```suggestion
  "mirage-crypto-rng" {>= "1.0.0"}
```

File: packages/websocket-lwt-unix/websocket-lwt-unix.2.17/opam [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+homepage: "https://github.com/vbmithr/ocaml-websocket"
+doc: "https://vbmithr.github.io/ocaml-websocket/doc"
+bug-reports: "https://github.com/vbmithr/ocaml-websocket/issues"
+depends: [
+  "dune" {>= "2.9"}
+  "ocaml" {>= "4.06.0"}
+  "websocket" {= version}
+  "lwt_log" {>= "1.1.1"}
+  "cohttp-lwt-unix" {>= "5.0.0"}
+  "odoc" {with-doc}
```
Comment (mseri) [inline packages/websocket-lwt-unix/websocket-lwt-unix.2.17/opam:30]:
```suggestion
  "sexplib"
  "odoc" {with-doc}
```
Due to
```
#=== ERROR while compiling websocket-lwt-unix.2.17 ============================#
# context              2.3.0~alpha~dev | linux/x86_64 | ocaml-base-compiler.4.14.2 | file:///home/opam/opam-repository
# path                 ~/.opam/4.14/.opam-switch/build/websocket-lwt-unix.2.17
# command              ~/.opam/opam-init/hooks/sandbox.sh build dune build -p websocket-lwt-unix -j 255 --promote-install-files=false @install
# exit-code            1
# env-file             ~/.opam/log/websocket-lwt-unix-7-c68292.env
# output-file          ~/.opam/log/websocket-lwt-unix-7-c68292.out
### output ###
# File "lwt/dune", line 20, characters 48-55:
# 20 |  (libraries conduit-lwt-unix websocket-lwt-unix sexplib))
```

File: packages/websocket-async/websocket-async.2.17/opam [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+homepage: "https://github.com/vbmithr/ocaml-websocket"
+doc: "https://vbmithr.github.io/ocaml-websocket/doc"
+bug-reports: "https://github.com/vbmithr/ocaml-websocket/issues"
+depends: [
+  "dune" {>= "2.9"}
+  "ocaml" {>= "4.06.0"}
+  "websocket" {= version}
+  "core" {>= "v0.15.0"}
+  "core_unix" {>= "v0.15.0"}
+  "async" {>= "v0.15.0"}
```
Comment (mseri) [inline packages/websocket-async/websocket-async.2.17/opam:30]:
```suggestion
  "async" {>= "v0.17.0"}
```
Due to
```
#=== ERROR while compiling websocket-async.2.17 ===============================#
# context              2.3.0~alpha~dev | linux/x86_64 | ocaml-base-compiler.4.14.2 | pinned(https://github.com/vbmithr/ocaml-websocket/releases/download/2.17/websocket-2.17.tbz)
# path                 ~/.opam/4.14/.opam-switch/build/websocket-async.2.17
# command              ~/.opam/opam-init/hooks/sandbox.sh build dune build -p websocket-async -j 39 --promote-install-files=false @install
# exit-code            1
# env-file             ~/.opam/log/websocket-async-7-41d7d9.env
# output-file          ~/.opam/log/websocket-async-7-41d7d9.out
### output ###
# (cd _build/default && /home/opam/.opam/4.14/bin/ocamlopt.opt -w -40 -g -I async/.websocket_async.objs/byte -I async/.websocket_async.objs/native -I /home/opam/.opam/4.14/lib/angstrom -I /home/opam/.opam/4.14/lib/astring -I /home/opam/.opam/4.14/lib/async -I /home/opam/.opam/4.14/lib/async/async_command -I /home/opam/.opam/4.14/lib/async/async_quickcheck -I /home/opam/.opam/4.14/lib/async/async_rpc -I /home/opam/.opam/4.14/lib/async_kernel -I /home/opam/.opam/4.14/lib/async_kernel/config -I /home/opam/.opam/4.14/lib/async_kernel/eager_deferred -I /home/opam/.opam/4.14/lib/async_kernel/persistent_connection_kernel -I /home/opam/.opam/4.14/lib/async_kernel/read_write_pair -I /home/opam/.opam/4.14/lib/async_rpc_kernel -I /home/opam/.opam/4.14/lib/async_unix -I /home/opam/.opam/4.14/lib/async_unix/thread_pool -I /home/opam/.opam/4.14/lib/async_unix/thread_safe_ivar -I /home/opam/.opam/4.14/lib/base -I /home/opam/.opam/4.14/lib/base/base_internalhash_types -I /home/opam/.opam/4.14/lib/base/caml -I /home/opam/.opam/4.14/lib/base/md5 -I /home/opam/.opam/4.14/lib/base/shadow_stdlib -I /home/opam/.opam/4.14/lib/base64 -I /home/opam/.opam/4.14/lib/base_bigstring -I /home/opam/.opam/4.14/lib/base_quickcheck -I /home/opam/.opam/4.14/lib/base_quickcheck/ppx_quickcheck/runtime -I /home/opam/.opam/4.14/lib/bigstringaf -I /home/opam/.opam/4.14/lib/bin_prot -I /home/opam/.opam/4.14/lib/bin_prot/shape -I /home/opam/.opam/4.14/lib/bytes -I /home/opam/.opam/4.14/lib/cohttp -I /home/opam/.opam/4.14/lib/cohttp-async -I /home/opam/.opam/4.14/lib/conduit -I /home/opam/.opam/4.14/lib/conduit-async -I /home/opam/.opam/4.14/lib/core -I /home/opam/.opam/4.14/lib/core/base_for_tests -I /home/opam/.opam/4.14/lib/core/command -I /home/opam/.opam/4.14/lib/core/filename_base -I /home/opam/.opam/4.14/lib/core/heap_block -I /home/opam/.opam/4.14/lib/core/univ_map -I /home/opam/.opam/4.14/lib/core/validate -I /home/opam/.opam/4.14/lib/core_kernel/bounded_int_table -I /home/opam/.opam/4.14/lib/core_kernel/bus -I /home/opam/.opam/4.14/lib/core_kernel/caml_threads -I /home/opam/.opam/4.14/lib/core_kernel/caml_unix -I /home/opam/.opam/4.14/lib/core_kernel/flags -I /home/opam/.opam/4.14/lib/core_kernel/iobuf -I /home/opam/.opam/4.14/lib/core_kernel/moption -I /home/opam/.opam/4.14/lib/core_kernel/pairing_heap -I /home/opam/.opam/4.14/lib/core_kernel/sexp_hidden_in_test -I /home/opam/.opam/4.14/lib/core_kernel/thread_pool_cpu_affinity -I /home/opam/.opam/4.14/lib/core_kernel/thread_safe_queue -I /home/opam/.opam/4.14/lib/core_kernel/timing_wheel -I /home/opam/.opam/4.14/lib/core_kernel/tuple_pool -I /home/opam/.opam/4.14/lib/core_kernel/uopt -I /home/opam/.opam/4.14/lib/core_kernel/uuid -I /home/opam/.opam/4.14/lib/core_unix -I /home/opam/.opam/4.14/lib/core_unix/bigstring_unix -I /home/opam/.opam/4.14/lib/core_unix/core_thread -I /home/opam/.opam/4.14/lib/core_unix/error_checking_mutex -I /home/opam/.opam/4.14/lib/core_unix/filename_unix -I /home/opam/.opam/4.14/lib/core_unix/iobuf_unix -I /home/opam/.opam/4.14/lib/core_unix/linux_ext -I /home/opam/.opam/4.14/lib/core_unix/nano_mutex -I /home/opam/.opam/4.14/lib/core_unix/ocaml_c_utils -I /home/opam/.opam/4.14/lib/core_unix/signal_unix -I /home/opam/.opam/4.14/lib/core_unix/squeue -I /home/opam/.opam/4.14/lib/core_unix/sys_unix -I /home/opam/.opam/4.14/lib/core_unix/time_float_unix -I /home/opam/.opam/4.14/lib/core_unix/time_ns_unix -I /home/opam/.opam/4.14/lib/core_unix/time_stamp_counter -I /home/opam/.opam/4.14/lib/core_unix/uuid -I /home/opam/.opam/4.14/lib/digestif -I /home/opam/.opam/4.14/lib/domain-name -I /home/opam/.opam/4.14/lib/eqaf -I /home/opam/.opam/4.14/lib/fieldslib -I /home/opam/.opam/4.14/lib/fmt -I /home/opam/.opam/4.14/lib/int_repr -I /home/opam/.opam/4.14/lib/ipaddr -I /home/opam/.opam/4.14/lib/ipaddr-sexp -I /home/opam/.opam/4.14/lib/ipaddr/unix -I /home/opam/.opam/4.14/lib/jane-street-headers -I /home/opam/.opam/4.14/lib/logs -I /home/opam/.opam/4.14/lib/logs-async -I /home/opam/.opam/4.14/lib/macaddr -I /home/opam/.opam/4.14/lib/magic-mime -I /home/opam/.opam/4.14/lib/mirage-crypto -I /home/opam/.opam/4.14/lib/mirage-crypto-rng -I /home/opam/.opam/4.14/lib/ocaml/threads -I /home/opam/.opam/4.14/lib/ocaml_intrinsics -I /home/opam/.opam/4.14/lib/ocplib-endian -I /home/opam/.opam/4.14/lib/parsexp -I /home/opam/.opam/4.14/lib/ppx_assert/runtime-lib -I /home/opam/.opam/4.14/lib/ppx_bench/runtime-lib -I /home/opam/.opam/4.14/lib/ppx_compare/runtime-lib -I /home/opam/.opam/4.14/lib/ppx_enumerate/runtime-lib -I /home/opam/.opam/4.14/lib/ppx_expect/collector -I /home/opam/.opam/4.14/lib/ppx_expect/common -I /home/opam/.opam/4.14/lib/ppx_expect/config -I /home/opam/.opam/4.14/lib/ppx_expect/config_types -I /home/opam/.opam/4.14/lib/ppx_hash/runtime-lib -I /home/opam/.opam/4.14/lib/ppx_here/runtime-lib -I /home/opam/.opam/4.14/lib/ppx_inline_test/config -I /home/opam/.opam/4.14/lib/ppx_inline_test/runtime-lib -I /home/opam/.opam/4.14/lib/ppx_log/types -I /home/opam/.opam/4.14/lib/ppx_module_timer/runtime -I /home/opam/.opam/4.14/lib/ppx_sexp_conv/runtime-lib -I /home/opam/.opam/4.14/lib/ppx_stable_witness/runtime -I /home/opam/.opam/4.14/lib/ppx_stable_witness/stable_witness -I /home/opam/.opam/4.14/lib/protocol_version_header -I /home/opam/.opam/4.14/lib/re -I /home/opam/.opam/4.14/lib/seq -I /home/opam/.opam/4.14/lib/sexplib -I /home/opam/.opam/4.14/lib/sexplib/unix -I /home/opam/.opam/4.14/lib/sexplib0 -I /home/opam/.opam/4.14/lib/spawn -I /home/opam/.opam/4.14/lib/splittable_random -I /home/opam/.opam/4.14/lib/stdio -I /home/opam/.opam/4.14/lib/stringext -I /home/opam/.opam/4.14/lib/time_now -I /home/opam/.opam/4.14/lib/timezone -I /home/opam/.opam/4.14/lib/typerep -I /home/opam/.opam/4.14/lib/uri -I /home/opam/.opam/4.14/lib/uri-sexp -I /home/opam/.opam/4.14/lib/uri/services -I /home/opam/.opam/4.14/lib/variantslib -I /home/opam/.opam/4.14/lib/websocket -intf-suffix .ml -no-alias-deps -o async/.websocket_async.objs/native/websocket_async.cmx -c -impl async/websocket_async.ml)
# File "async/websocket_async.ml", line 86, characters 44-57:
# 86 |     Option.iter initialized ~f:(fun ivar -> Ivar.fill_exn ivar ());
#                                                  ^^^^^^^^^^^^^
# Error: Unbound value Ivar.fill_exn
# (cd _build/default && /home/opam/.opam/4.14/bin/ocamlc.opt -w -40 -g -bin-annot -I async/.websocket_async.objs/byte -I /home/opam/.opam/4.14/lib/angstrom -I /home/opam/.opam/4.14/lib/astring -I /home/opam/.opam/4.14/lib/async -I /home/opam/.opam/4.14/lib/async/async_command -I /home/opam/.opam/4.14/lib/async/async_quickcheck -I /home/opam/.opam/4.14/lib/async/async_rpc -I /home/opam/.opam/4.14/lib/async_kernel -I /home/opam/.opam/4.14/lib/async_kernel/config -I /home/opam/.opam/4.14/lib/async_kernel/eager_deferred -I /home/opam/.opam/4.14/lib/async_kernel/persistent_connection_kernel -I /home/opam/.opam/4.14/lib/async_kernel/read_write_pair -I /home/opam/.opam/4.14/lib/async_rpc_kernel -I /home/opam/.opam/4.14/lib/async_unix -I /home/opam/.opam/4.14/lib/async_unix/thread_pool -I /home/opam/.opam/4.14/lib/async_unix/thread_safe_ivar -I /home/opam/.opam/4.14/lib/base -I /home/opam/.opam/4.14/lib/base/base_internalhash_types -I /home/opam/.opam/4.14/lib/base/caml -I /home/opam/.opam/4.14/lib/base/md5 -I /home/opam/.opam/4.14/lib/base/shadow_stdlib -I /home/opam/.opam/4.14/lib/base64 -I /home/opam/.opam/4.14/lib/base_bigstring -I /home/opam/.opam/4.14/lib/base_quickcheck -I /home/opam/.opam/4.14/lib/base_quickcheck/ppx_quickcheck/runtime -I /home/opam/.opam/4.14/lib/bigstringaf -I /home/opam/.opam/4.14/lib/bin_prot -I /home/opam/.opam/4.14/lib/bin_prot/shape -I /home/opam/.opam/4.14/lib/bytes -I /home/opam/.opam/4.14/lib/cohttp -I /home/opam/.opam/4.14/lib/cohttp-async -I /home/opam/.opam/4.14/lib/conduit -I /home/opam/.opam/4.14/lib/conduit-async -I /home/opam/.opam/4.14/lib/core -I /home/opam/.opam/4.14/lib/core/base_for_tests -I /home/opam/.opam/4.14/lib/core/command -I /home/opam/.opam/4.14/lib/core/filename_base -I /home/opam/.opam/4.14/lib/core/heap_block -I /home/opam/.opam/4.14/lib/core/univ_map -I /home/opam/.opam/4.14/lib/core/validate -I /home/opam/.opam/4.14/lib/core_kernel/bounded_int_table -I /home/opam/.opam/4.14/lib/core_kernel/bus -I /home/opam/.opam/4.14/lib/core_kernel/caml_threads -I /home/opam/.opam/4.14/lib/core_kernel/caml_unix -I /home/opam/.opam/4.14/lib/core_kernel/flags -I /home/opam/.opam/4.14/lib/core_kernel/iobuf -I /home/opam/.opam/4.14/lib/core_kernel/moption -I /home/opam/.opam/4.14/lib/core_kernel/pairing_heap -I /home/opam/.opam/4.14/lib/core_kernel/sexp_hidden_in_test -I /home/opam/.opam/4.14/lib/core_kernel/thread_pool_cpu_affinity -I /home/opam/.opam/4.14/lib/core_kernel/thread_safe_queue -I /home/opam/.opam/4.14/lib/core_kernel/timing_wheel -I /home/opam/.opam/4.14/lib/core_kernel/tuple_pool -I /home/opam/.opam/4.14/lib/core_kernel/uopt -I /home/opam/.opam/4.14/lib/core_kernel/uuid -I /home/opam/.opam/4.14/lib/core_unix -I /home/opam/.opam/4.14/lib/core_unix/bigstring_unix -I /home/opam/.opam/4.14/lib/core_unix/core_thread -I /home/opam/.opam/4.14/lib/core_unix/error_checking_mutex -I /home/opam/.opam/4.14/lib/core_unix/filename_unix -I /home/opam/.opam/4.14/lib/core_unix/iobuf_unix -I /home/opam/.opam/4.14/lib/core_unix/linux_ext -I /home/opam/.opam/4.14/lib/core_unix/nano_mutex -I /home/opam/.opam/4.14/lib/core_unix/ocaml_c_utils -I /home/opam/.opam/4.14/lib/core_unix/signal_unix -I /home/opam/.opam/4.14/lib/core_unix/squeue -I /home/opam/.opam/4.14/lib/core_unix/sys_unix -I /home/opam/.opam/4.14/lib/core_unix/time_float_unix -I /home/opam/.opam/4.14/lib/core_unix/time_ns_unix -I /home/opam/.opam/4.14/lib/core_unix/time_stamp_counter -I /home/opam/.opam/4.14/lib/core_unix/uuid -I /home/opam/.opam/4.14/lib/digestif -I /home/opam/.opam/4.14/lib/domain-name -I /home/opam/.opam/4.14/lib/eqaf -I /home/opam/.opam/4.14/lib/fieldslib -I /home/opam/.opam/4.14/lib/fmt -I /home/opam/.opam/4.14/lib/int_repr -I /home/opam/.opam/4.14/lib/ipaddr -I /home/opam/.opam/4.14/lib/ipaddr-sexp -I /home/opam/.opam/4.14/lib/ipaddr/unix -I /home/opam/.opam/4.14/lib/jane-street-headers -I /home/opam/.opam/4.14/lib/logs -I /home/opam/.opam/4.14/lib/logs-async -I /home/opam/.opam/4.14/lib/macaddr -I /home/opam/.opam/4.14/lib/magic-mime -I /home/opam/.opam/4.14/lib/mirage-crypto -I /home/opam/.opam/4.14/lib/mirage-crypto-rng -I /home/opam/.opam/4.14/lib/ocaml/threads -I /home/opam/.opam/4.14/lib/ocaml_intrinsics -I /home/opam/.opam/4.14/lib/ocplib-endian -I /home/opam/.opam/4.14/lib/parsexp -I /home/opam/.opam/4.14/lib/ppx_assert/runtime-lib -I /home/opam/.opam/4.14/lib/ppx_bench/runtime-lib -I /home/opam/.opam/4.14/lib/ppx_compare/runtime-lib -I /home/opam/.opam/4.14/lib/ppx_enumerate/runtime-lib -I /home/opam/.opam/4.14/lib/ppx_expect/collector -I /home/opam/.opam/4.14/lib/ppx_expect/common -I /home/opam/.opam/4.14/lib/ppx_expect/config -I /home/opam/.opam/4.14/lib/ppx_expect/config_types -I /home/opam/.opam/4.14/lib/ppx_hash/runtime-lib -I /home/opam/.opam/4.14/lib/ppx_here/runtime-lib -I /home/opam/.opam/4.14/lib/ppx_inline_test/config -I /home/opam/.opam/4.14/lib/ppx_inline_test/runtime-lib -I /home/opam/.opam/4.14/lib/ppx_log/types -I /home/opam/.opam/4.14/lib/ppx_module_timer/runtime -I /home/opam/.opam/4.14/lib/ppx_sexp_conv/runtime-lib -I /home/opam/.opam/4.14/lib/ppx_stable_witness/runtime -I /home/opam/.opam/4.14/lib/ppx_stable_witness/stable_witness -I /home/opam/.opam/4.14/lib/protocol_version_header -I /home/opam/.opam/4.14/lib/re -I /home/opam/.opam/4.14/lib/seq -I /home/opam/.opam/4.14/lib/sexplib -I /home/opam/.opam/4.14/lib/sexplib/unix -I /home/opam/.opam/4.14/lib/sexplib0 -I /home/opam/.opam/4.14/lib/spawn -I /home/opam/.opam/4.14/lib/splittable_random -I /home/opam/.opam/4.14/lib/stdio -I /home/opam/.opam/4.14/lib/stringext -I /home/opam/.opam/4.14/lib/time_now -I /home/opam/.opam/4.14/lib/timezone -I /home/opam/.opam/4.14/lib/typerep -I /home/opam/.opam/4.14/lib/uri -I /home/opam/.opam/4.14/lib/uri-sexp -I /home/opam/.opam/4.14/lib/uri/services -I /home/opam/.opam/4.14/lib/variantslib -I /home/opam/.opam/4.14/lib/websocket -intf-suffix .ml -no-alias-deps -o async/.websocket_async.objs/byte/websocket_async.cmo -c -impl async/websocket_async.ml)
# File "async/websocket_async.ml", line 86, characters 44-57:
# 86 |     Option.iter initialized ~f:(fun ivar -> Ivar.fill_exn ivar ());
#                                                  ^^^^^^^^^^^^^
# Error: Unbound value Ivar.fill_exn
```

File: packages/websocket-lwt-unix/websocket-lwt-unix.2.17/opam [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+tags: ["org:mirage" "org:xapi-project"]
+homepage: "https://github.com/vbmithr/ocaml-websocket"
+doc: "https://vbmithr.github.io/ocaml-websocket/doc"
+bug-reports: "https://github.com/vbmithr/ocaml-websocket/issues"
+depends: [
+  "dune" {>= "2.9"}
+  "ocaml" {>= "4.06.0"}
+  "websocket" {= version}
+  "lwt_log" {>= "1.1.1"}
+  "cohttp-lwt-unix" {>= "5.0.0"}
```
Comment (mseri) [inline packages/websocket-lwt-unix/websocket-lwt-unix.2.17/opam:29]:
```suggestion
  "cohttp-lwt-unix" {>= "5.0.0" & < "6.0.0~"}
```

Comment (mseri) [reply packages/websocket-lwt-unix/websocket-lwt-unix.2.17/opam]: Due to
```
# File "lwt/websocket_lwt_unix.ml", line 87, characters 35-37:
# 87 |     (fun () -> drain_handshake req ic oc nonce)
#                                         ^^
# Error: This expression has type
#          Conduit_lwt_unix.ic = Lwt_io.input Lwt_io.channel
#        but an expression was expected of type
#          Response.IO.ic = Cohttp_lwt_unix__Input_channel.t
```

**Comment by mseri** [comment]
Looks fine now, thanks

**Comment by mseri** [comment]
@vbmithr please make these fixes upstream

**Pull request merged**